adapters/amqp-vertx/src/test/java/org/eclipse/hono/adapter/amqp/impl/VertxBasedAmqpProtocolAdapterTest.java

```diff
@@ -1189,10 +1191,17 @@ public void testConnectionFailsIfAdapterLevelConnectionLimitIsExceeded() {
                 .forClass(Handler.class);
         verify(deviceConnection).openHandler(openHandler.capture());
         openHandler.getValue().handle(Future.succeededFuture(deviceConnection));
-        // THEN the adapter does not accept the incoming connection request. 
+        // THEN the adapter closes the connection right after it opened it
+        final ArgumentCaptor<Handler<AsyncResult<ProtonConnection>>> closeHandler = ArgumentCaptor.forClass(Handler.class);
+        verify(deviceConnection).closeHandler(closeHandler.capture());
+        closeHandler.getValue().handle(Future.succeededFuture());
         final ArgumentCaptor<ErrorCondition> errorConditionCaptor = ArgumentCaptor.forClass(ErrorCondition.class);
         verify(deviceConnection).setCondition(errorConditionCaptor.capture());
         assertEquals(AmqpError.UNAUTHORIZED_ACCESS, errorConditionCaptor.getValue().getCondition());
+        // AND increments and decrements the connection count accordingly
+        final InOrder metricsInOrderVerifier = inOrder(metrics);
+        metricsInOrderVerifier.verify(metrics).incrementConnections(TEST_TENANT_ID);
```

Comment:
FMPOV this check should be done immediately after the openHandler has been invoked

Reply:
Okay for me. By this we'll make sure that the increment is actually made in the open handler and not accidentally called in the closed handler together with the decrement (would be very weird - but you never now... )

Reply:
exactly